website/en/docs/faq/index.md

```diff
@@ -143,6 +143,76 @@ const isNumber = (valueToRefine: ?number): %checks => typeof valueToRefine === '
 if (isNumber(val)) add(val, 2);
 ```
 
+### Why can't I pass Array<string> to function that takes Array<string | number>
+
+The function allows string values in an array, but in this case Flow protects the original array. In the function you would be able to push number to it which means that the type of original array would not be true. You can get rid of this by changing the argument to `$ReadOnlyArray<string | number>`. This prevents following code to push anything which makes it safe to pass stricter types to it.
+
+This would not work:
+
+```js
+// @flow
+
+const fn = (arr: Array<string | number>) => {
+	// arr.push(123) // note! Array<string> passed in and after this it would also contain numbers
+	return arr
+};
+
+const arr: Array<string> = ['abc'];
+
+fn(arr); // Error!
+```
+
+but with $ReadOnlyArray you can achieve what you were looking for:
+
+```js
+// @flow
+const fn = (arr: $ReadOnlyArray<string | number>) => {
+    // arr.push(321) //note! Array<string> passed in and after this it would also contain numbers
+	return arr;
+};
+
+const arr: Array<string> = ['abc'];
+
+fn(arr);
+```
+
+Example ([https://flow.org/try](https://flow.org/try/#0PTAEAEDMBsHsHcBQiSgOoEsAuALWBXLUAJwFMBDAE1gDtoBPALmQGNaBnIyGgRlAF5QACnLFijUAEEx5egB5OxDDQDmoAD4ACmvgC2AI1LEAfAEoBx0AG9EASFSjiAOgAO+djiE8ATAGZzIDSwWKQAhFIy8orKKpYu5OzspJSgyqDkNCnkkCHEoLgY7KlE8ATQWdDssKBsNFjkaToGRux2ZFj4xDTpYogAvqwcRI48EtLEsgpYSqqWggDaAOTk+iyLALoA3Mio3DwiYqaboKhyALQXbLq6pHWp3ZCweUbET+yhoTtgFz+-f-8AwFAr6gABKFGodHoZwAbi0MLRmIhapxQNxvAJhI4JAAScFUADyUPGk2iqg02j0hhM5n4lhsoFA9jAjlc7k8vm8PACwCCIXCJKi0xicQSSRSaQyWRyRnyOEKxVApXw5XSlWqtXqjSpLTapA6XR6xH6gxoqMc3jGkSmM1imKWKzWW2Q6IOxG8RyAA))
+
+### Why can't I pass { a: string } to function that takes { a: string | number }
+
+The function allows string values in an object, but in this case Flow protects the original object: In the function you would be able to mutate the object so that the property `a` would have number in it which means that the type of original object would not be true anymore. You can get rid of this by making the property covariant (read-only) `{ +a: string | number }`. This prevents following code to change the property which makes it safe to pass stricter types to the function.
+
+This would not work:
+
+```js
+// @flow
+
+const fn1 = (obj: {| a: string | number |}) => {
+	// obj.a = 123 //note! {| a: string |} passed in and after this it would also contain numbers
+	return obj;
+};
+
+const object: {| a: string |} = {a: 'str' };
+
+fn(object); // Error!
+```
+
+but with covariant property you can achieve what you were looking for:
+
+```js
+// @flow
+const fn = (obj: {| +a: string | number |}) => {
+  	// obj.a = 123 //note! {| a: string |} passed in and after this it would also contain numbers
```

Comment:
Here to, it's interesting to point out in the first example why it's an error, and here why the types prevent us from doing this error!

Reply:
Good points!